lib/ansible/utils/__init__.py

```diff
@@ -1104,18 +1105,26 @@ def ask_passwords(ask_pass=False, ask_sudo_pass=False, ask_su_pass=False, ask_va
 
     if ask_pass:
         sshpass = getpass.getpass(prompt="SSH password: ")
+        if sshpass:
+            sshpass = to_bytes(sshpass, errors='strict', nonstring='simplerepr')
         sudo_prompt = "sudo password [defaults to SSH password]: "
 
     if ask_sudo_pass:
         sudopass = getpass.getpass(prompt=sudo_prompt)
         if ask_pass and sudopass == '':
             sudopass = sshpass
+        if sudopass:
+            sudopass = to_bytes(sudopass, errors='strict', nonstring='simplerepr')
 
     if ask_su_pass:
         su_pass = getpass.getpass(prompt=su_prompt)
+        if supass:
+            supass = to_bytes(supass, errors='strict', nonstring='simplerepr')
```

Comment:
```su_pass``` is used in the getpass but ```supass``` without the underscore is used here.  Probably needs to be ```su_pass``` in both places.